cep-recipe-extensions.md

```diff
@@ -0,0 +1,54 @@
+# Extensions to the recipe spec
+
+This document describes extensions to the recipe specification as merged in cep-14.
+
+## The `build.post_process` section
+
+We became aware that the `conda_build_config.yaml` has ways to specify post-processing steps. These are regex replacements that are performed on any new files. 
+
+Instead of adding these instructions to the `conda_build_config.yaml`, we decided to add a new section to the recipe spec: `build.post_process`. This section is a list of dictionaries, each with the following keys:
+
+- `files`: globs to select files to process
+- `regex`: the regex to apply
+- `replacement`: the replacement string
+
+The regex specification follows Rust `regex` syntax. Most notably, the replacement string can refer to capture groups with `$1`, `$2`, etc.
+This also means that replacement strings need to escape `$` with `$$`.
+
+Internally, we use `replace_all` from the `regex` crate. This means that the regex is applied to the entire file, not line by line.
+
+### Example
+
+```yaml
+build:
+  post_process:
+    - files:
+        - "*.txt"
+      regex: "foo"
+      replacement: "bar"
+    - files:
+        - "*.cmake"
+      regex: "/sysroot/"
+      replacement: "$${PREFIX}/"
+```
+
+## Globs, positive and negative
```

Comment:
Does this have to do with recipe post-processing? 🤔

Reply:
Yeah. This really feels like it should just be incorporated into the build script(s).

Reply:
The positive/negative globs should be part of the recipe. Using the build scripts just to shuffle files around into separate output is a lot of core for little gain, especially because it's a well-defined operation and doesn't overload the recipe syntax IMO.

C.f. also https://github.com/conda/conda-build/pull/5216

Reply:
No this doesn't really have anything to do with post-processing per se. The idea is that any list of globs can also include negative globs. 

List of globs appear in multiple places in the recipe.